# Sleep health and lifestyle

In this project we took the Sleep Health and Lifestyle Dataset in where you can find data on different people such as their profession, physical activity and sleep quality. We will take this dataset to know what are the things that affect sleep such as the amount of physical activity, the profession and the pressure to which each person is subjected.

## Import libraries

In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

## Load data

In [2]:
data = pd.read_csv(
    "/kaggle/input/sleep-health-and-lifestyle-dataset/Sleep_health_and_lifestyle_dataset.csv",
    index_col = "Person ID"
)
data.head()

Gender  Age            Occupation  Sleep Duration  Quality of Sleep  \
Person ID                                                                       
1           Male   27     Software Engineer             6.1                 6   
2           Male   28                Doctor             6.2                 6   
3           Male   28                Doctor             6.2                 6   
4           Male   28  Sales Representative             5.9                 4   
5           Male   28  Sales Representative             5.9                 4   

           Physical Activity Level  Stress Level BMI Category Blood Pressure  \
Person ID                                                                      
1                               42             6   Overweight         126/83   
2                               60             8       Normal         125/80   
3                               60             8       Normal         125/80   
4                               30             8        Obese         140/90   
5                               30             8        Obese         140/90   

           Heart Rate  Daily Steps Sleep Disorder  
Person ID                                          
1                  77         4200            NaN  
2                  75        10000            NaN  
3                  75        10000            NaN  
4                  85         3000    Sleep Apnea  
5                  85         3000    Sleep Apnea

## Questions for analysis

- Who has worse sleep quality, men or women?
- Is there a relationship between people's sleep quality and their profession?
- Does physical activity affect sleep?
- Which profession has the worst sleep quality?
- Which profession have people with the highest level of stress and the highest body mass index?
- In what age range are the most sleep disorders found?
- Does the number of steps per day affect sleep quality and body mass index?

Summary of the data set (using `.info()` method)

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 374 entries, 1 to 374
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Gender                   374 non-null    object 
 1   Age                      374 non-null    int64  
 2   Occupation               374 non-null    object 
 3   Sleep Duration           374 non-null    float64
 4   Quality of Sleep         374 non-null    int64  
 5   Physical Activity Level  374 non-null    int64  
 6   Stress Level             374 non-null    int64  
 7   BMI Category             374 non-null    object 
 8   Blood Pressure           374 non-null    object 
 9   Heart Rate               374 non-null    int64  
 10  Daily Steps              374 non-null    int64  
 11  Sleep Disorder           155 non-null    object 
dtypes: float64(1), int64(6), object(5)
memory usage: 38.0+ KB


Generate descriptive statistics (using `.describe()` method).

Descriptive statistics include those that summarize the central tendency, dispersion and shape of a dataset's distribution, excluding NaN values.

In [4]:
data.describe()

Age  Sleep Duration  Quality of Sleep  Physical Activity Level  \
count  374.000000      374.000000        374.000000               374.000000   
mean    42.184492        7.132086          7.312834                59.171123   
std      8.673133        0.795657          1.196956                20.830804   
min     27.000000        5.800000          4.000000                30.000000   
25%     35.250000        6.400000          6.000000                45.000000   
50%     43.000000        7.200000          7.000000                60.000000   
75%     50.000000        7.800000          8.000000                75.000000   
max     59.000000        8.500000          9.000000                90.000000   

       Stress Level  Heart Rate   Daily Steps  
count    374.000000  374.000000    374.000000  
mean       5.385027   70.165775   6816.844920  
std        1.774526    4.135676   1617.915679  
min        3.000000   65.000000   3000.000000  
25%        4.000000   68.000000   5600.000000  
50%        5.000000   70.000000   7000.000000  
75%        7.000000   72.000000   8000.000000  
max        8.000000   86.000000  10000.000000

## Ages analisys

Count the number of men and women

In [5]:
data["Gender"].value_counts()

# or

# data.Gender.value_counts()

Gender
Male      189
Female    185
Name: count, dtype: int64

> This gives us a general overview of the group, and we can say that the group is in proportion to the number of men and women. Proportion, with respect to the number of men and women.

## Classification by age

In the set we can find the ages of the persons, with respect to the ages we can find the following:

In [6]:
data["Age"].min()

27

In [7]:
data["Age"].max()

59

The lowest age is 27 years old
The highest age is 59 years old

What we can say is that the data set takes into account people who are already of working age, starting work or with only a few years in it, or people close to retiring. Let's find the average age in the data set.

In [8]:
data["Age"].mean()

42.18449197860963

The average age of the data set is 42 years and the standard deviation is 8.67, which means that in this set we can find very dispersed data, which may be due to the few data collected.

To support us in this analysis, a new column was added to the data set called Age_group in which the age of the people was classified as follows:

- **Age less than or equal to 40 years**: *Adult*.
- **Age over 40 years**: *OlderAdult*

In [9]:
data.loc[data["Age"] <= 40, 'AgeGroup'] = "Adult"
data.loc[data["Age"] >= 41, 'AgeGroup'] = "OlderAdult"

# View the first rows of the data set
data.head()

Gender  Age            Occupation  Sleep Duration  Quality of Sleep  \
Person ID                                                                       
1           Male   27     Software Engineer             6.1                 6   
2           Male   28                Doctor             6.2                 6   
3           Male   28                Doctor             6.2                 6   
4           Male   28  Sales Representative             5.9                 4   
5           Male   28  Sales Representative             5.9                 4   

           Physical Activity Level  Stress Level BMI Category Blood Pressure  \
Person ID                                                                      
1                               42             6   Overweight         126/83   
2                               60             8       Normal         125/80   
3                               60             8       Normal         125/80   
4                               30             8        Obese         140/90   
5                               30             8        Obese         140/90   

           Heart Rate  Daily Steps Sleep Disorder AgeGroup  
Person ID                                                   
1                  77         4200            NaN    Adult  
2                  75        10000            NaN    Adult  
3                  75        10000            NaN    Adult  
4                  85         3000    Sleep Apnea    Adult  
5                  85         3000    Sleep Apnea    Adult

In [10]:
# View the last rows of the data set
data.tail()

Gender  Age Occupation  Sleep Duration  Quality of Sleep  \
Person ID                                                             
370        Female   59      Nurse             8.1                 9   
371        Female   59      Nurse             8.0                 9   
372        Female   59      Nurse             8.1                 9   
373        Female   59      Nurse             8.1                 9   
374        Female   59      Nurse             8.1                 9   

           Physical Activity Level  Stress Level BMI Category Blood Pressure  \
Person ID                                                                      
370                             75             3   Overweight         140/95   
371                             75             3   Overweight         140/95   
372                             75             3   Overweight         140/95   
373                             75             3   Overweight         140/95   
374                             75             3   Overweight         140/95   

           Heart Rate  Daily Steps Sleep Disorder    AgeGroup  
Person ID                                                      
370                68         7000    Sleep Apnea  OlderAdult  
371                68         7000    Sleep Apnea  OlderAdult  
372                68         7000    Sleep Apnea  OlderAdult  
373                68         7000    Sleep Apnea  OlderAdult  
374                68         7000    Sleep Apnea  OlderAdult

Taking into account this ordering of the data, we have the following:

In [11]:
data['AgeGroup'].value_counts()

AgeGroup
OlderAdult    209
Adult         165
Name: count, dtype: int64

Finally, the average age for each gender is as follows:

In [12]:
data.groupby("Gender")["Age"].agg(["mean", "std"])

mean       std
Gender                     
Female  47.405405  8.093407
Male    37.074074  5.662006

We can see that there is a difference of 10 years in the average age of both genders, and we also note a large difference in the standard deviation, which is 3 years.